gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/OracleExtractor.java

```diff
@@ -0,0 +1,297 @@
+      List<Predicate> predicateList)
+      throws HighWatermarkException {
+    LOG.debug("Build query to get high watermark");
+    List<Command> commands = Lists.newArrayList();
+
+    String columnProjection = "max(" + Utils.getCoalesceColumnNames(watermarkColumn) + ")";
+    String watermarkFilter = this.concatPredicates(predicateList);
+    String query = this.getExtractSql();
+
+    if (StringUtils.isBlank(watermarkFilter)) {
+      watermarkFilter = "1=1";
+    }
+    query = query.replace(this.getOutputColumnProjection(), columnProjection)
+        .replace(ConfigurationKeys.DEFAULT_SOURCE_QUERYBASED_WATERMARK_PREDICATE_SYMBOL, watermarkFilter);
+
+    commands.add(JdbcExtractor.getCommand(query, JdbcCommand.JdbcCommandType.QUERY));
+    return commands;
+  }
+
+  @Override
+  String getDefaultExtractQuery(String schema, String entity) {
+    return "SELECT " + getOutputColumnProjection() + " FROM " + entity + "@" + schema;
+  }
+
+  @Override
+  public List<Command> getCountMetadata(String schema, String entity, WorkUnit workUnit, List<Predicate> predicateList)
+      throws RecordCountException {
+    List<Command> commands = Lists.newArrayList();
+
+    String columnProjection = "COUNT(1)";
+    String watermarkFilter = this.concatPredicates(predicateList);
+    String query = this.getExtractSql();
+
+    if (StringUtils.isBlank(watermarkFilter)) {
+      watermarkFilter = "1=1";
+    }
+    query = query.replace(this.getOutputColumnProjection(), columnProjection)
+        .replace(ConfigurationKeys.DEFAULT_SOURCE_QUERYBASED_WATERMARK_PREDICATE_SYMBOL, watermarkFilter);
+    commands.add(JdbcExtractor.getCommand(query, JdbcCommand.JdbcCommandType.QUERY));
+    return commands;
+  }
+
+  @Override
+  public List<Command> getDataMetadata(String schema, String entity, WorkUnit workUnit, List<Predicate> predicateList)
+      throws DataRecordException {
+    LOG.debug("Build query to extract data");
+    List<Command> commands = Lists.newArrayList();
+
+    String watermarkFilter = this.concatPredicates(predicateList);
+    String query = this.getExtractSql();
+    if (StringUtils.isBlank(watermarkFilter)) {
+      watermarkFilter = "1=1";
+    }
+
+    query = query.replace(ConfigurationKeys.DEFAULT_SOURCE_QUERYBASED_WATERMARK_PREDICATE_SYMBOL, watermarkFilter);
+    commands.add(JdbcExtractor.getCommand(query, JdbcCommand.JdbcCommandType.QUERY));
+
+    int fetchSize = workUnit.getPropAsInt(ConfigurationKeys.SOURCE_QUERYBASED_FETCH_SIZE, ConfigurationKeys.DEFAULT_SOURCE_QUERYBASED_JDBC_RESULTSET_FETCH_SIZE);
+    commands.add(JdbcExtractor.getCommand(fetchSize, JdbcCommand.JdbcCommandType.FETCHSIZE));
+    return commands;
+  }
+
+  @Override
+  public Map<String, String> getDataTypeMap() {
+    return DATA_TYPE_MAP;
+  }
+
+  @Override
+  public Iterator<JsonElement> getRecordSetFromSourceApi(String schema, String entity, WorkUnit workUnit,
+      List<Predicate> predicateList) throws IOException {
+    return null;
```

Comment:
Added comment on it (No need to support specific retrieval as getRecordSet in JdbcExtractor is sufficient) and changed it to throw exception.